### PR TITLE
Feature/add wishlist

### PR DESCRIPTION
**Pull request by bimal125**
- Depends on #3 

## Changes

* Add book in wish list
* Remove books from wish list

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [ ] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/book/models.py [hunk 1]
```diff
@@ -146,3 +146,21 @@ def save(self, *args, **kwargs):
             thumb_data = open(os.path.join('/tmp', filename), 'rb')
             self.og_image.save(filename, ImageFile(thumb_data), save=False)
             super(Book, self).save(*args, **kwargs)
+
+
+class WishList(models.Model):
+    user = models.ForeignKey(
```
Comment (thenav56) [inline apps/book/models.py]:
```suggestion
    created_by = models.ForeignKey(
```

File: apps/book/models.py [hunk 2]
```diff
@@ -146,3 +146,21 @@ def save(self, *args, **kwargs):
...
+        'user.User', verbose_name=_('User'), related_name='user',
```
Comment (thenav56) [inline apps/book/models.py]:
```suggestion
        'user.User', verbose_name=_('User'), related_name='wish_lists',
```

File: apps/book/models.py [hunk 3]
```diff
@@ -146,3 +146,21 @@ def save(self, *args, **kwargs):
...
+        on_delete=models.CASCADE
+    )
+    book = models.ForeignKey(
+        'book.Book', verbose_name=_('Book'), related_name='book',
```
Comment (thenav56) [inline apps/book/models.py]:
```suggestion
        'book.Book', verbose_name=_('Book'), related_name='+',
```

File: apps/book/serializers.py [hunk 4]
```diff
@@ -1,11 +1,26 @@
...
+    class Meta:
+        model = WishList
+        fields = ('book', )
+
+    def validate(self, attrs):
+        request = self.context.get('request')
+        attrs['user'] = request.user
+        book = attrs.get('book', None)
+        if WishList.objects.filter(user=request.user, book=book).exists():
+            raise serializers.ValidationError(_('Book is already added in wish list.'))
```
Comment (thenav56) [inline apps/book/serializers.py]: Use validate_book.

File: apps/book/serializers.py [hunk 5]
```diff
@@ -1,11 +1,26 @@
...
+
+class WishListSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = WishList
+        fields = ('book', )
+
+    def validate(self, attrs):
+        request = self.context.get('request')
+        attrs['user'] = request.user
```
Comment (thenav56) [inline apps/book/serializers.py]: Let's create a base class for auto set created_by and modified_by if they exists in the model.

**Review by thenav56 — CHANGES_REQUESTED** [review]

File: apps/book/mutations.py [hunk 6]
```diff
@@ -52,7 +52,44 @@ def check_permissions(cls, *args, **_):
...
+)
+
+
+class WishListMixin():
+    @classmethod
+    def filter_queryset(cls, qs, info):
+        return qs.filter(user=info.context.user)
+
+
+class CreateWishList(WishListMixin, CreateUpdateGrapheneMutation):
```
Comment (k9845) [inline apps/book/mutations.py:67]: Seems this performs both create and update, better name this `CreateUpdateWishList`

File: apps/book/schema.py [hunk 7]
```diff
@@ -92,3 +112,13 @@ class Query(graphene.ObjectType):
             page_size_query_param='pageSize'
         )
     )
+    wish_list = DjangoPaginatedListObjectField(
+        WishListListType,
+        pagination=PageGraphqlPagination(
+            page_size_query_param='pageSize'
+        )
+    )
```
Comment (k9845) [inline apps/book/schema.py:120]: Don't we require single wish_list query?

Comment (thenav56) [reply apps/book/schema.py:120]: Not required for now.

**Review by thenav56 — APPROVED** [review]

**Pull request merged**